src/common/base/src/runtime/global_runtime.rs

```diff
@@ -21,6 +21,15 @@ use crate::runtime::Runtime;
 
 pub struct GlobalIORuntime;
 
+pub struct GlobalQueryRuntime(pub Runtime);
+
+impl GlobalQueryRuntime {
+    #[inline(always)]
+    pub fn runtime<'a>(self: &'a Arc<Self>) -> &'a Runtime {
```

Comment:
Maybe `impl Deref` is better

Reply:
`Deref should only be implemented for smart pointers `

https://doc.rust-lang.org/std/ops/trait.Deref.html

Reply:
I will mark this discussion as solved and prepare to merge this PR.